### PR TITLE
Add Rust TLS Feature Flag For `opentelemetry-http`

### LINKED ISSUE #472: Rustls Support for Reqwest client
Is there any desire to support Rustls either by default or with a feature flag for the reqwest client used in these crates? It would be nice to not have to link OpenSSL.

### PR DESCRIPTION
**Pull request by LuisOsta**
Relates #472
Design discussion issue (if applicable) #

## Changes
Similar issue to what https://github.com/open-telemetry/opentelemetry-rust/issues/472 but for opentelemetry-http
Please provide a brief description of the changes here.
Added new feature flag for the `opentelemetry-http` crate to enable rustls instead of openssl

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Comment by LuisOsta** [comment]
Just signed the CLA

**Comment by LuisOsta** [comment]
@hdost @cijothomas What is the process necessary to get this merged? As far as I can tell the CI failures are unrelated to the changes made here

**Review by hdost — CHANGES_REQUESTED** [review]
Thanks for the work! Please add to CHANGELOG.

**Comment by LuisOsta** [comment]
@hdost Sounds good! Didn't know if this would be considered non-trivial or not. Will add to changelog

**Comment by LuisOsta** [comment]
@hdost Done!

**Review by hdost — APPROVED** [review]
Indeed it's certainly a trivial change, but this is something that others may indeed be interested in this.

**Comment by LuisOsta** [comment]
@hdost Sounds good, two questions:

- How do I go about merging this PR
- When should I expect the next release of the `opentelemetry-http` crate? And if its not soon do you have a recommended approach to accessing the changes?

File: opentelemetry-http/Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,9 @@ license = "Apache-2.0"
 edition = "2021"
 rust-version = "1.65"
 
+[features]
+reqwest-rustls = ["reqwest", "reqwest/rustls-tls"]
```
Comment (lalitb) [inline opentelemetry-http/Cargo.toml:13]: Just curious, in other parts of [code](https://github.com/open-telemetry/opentelemetry-rust/blob/815e3874a28f132415e7eead92385ff72ef2c9a2/opentelemetry-zipkin/Cargo.toml#L26), the feature depends on the `rustls-tls-native-roots`, here it is `rustls-tls`. The difference is the choice of the root-certificates crate. Does it makes any difference ?

Comment (lalitb) [reply opentelemetry-http/Cargo.toml:13]: Thanks @LuisOsta. I went through that discussion before commenting. The concern I had is - if the application is running in an environment that relies on enterprise-specific root certificates (such as those used for a mandatory proxy or internal CA), `webpki-roots` will not include these out of the box. The `rustls-tls-native-roots` looks more flexible in that regard. But I am not an expert here, so will leave it for others to comment.

Comment (LuisOsta) [reply opentelemetry-http/Cargo.toml:13]: @lalitb That's very fair, honestly seems like both have their tradeoffs. I'll switch to `rustls-tls-native-roots` to remain consistent then

**Review by lalitb — APPROVED** [review]
Thanks.

**Comment by LuisOsta** [comment]
@lalitb Do you know how I can merge the PR?

**Comment by lalitb** [comment]
> @lalitb Do you know how I can merge the PR? Even though its approved I can't since I'm not an authorized user

@LuisOsta This would be merged by one of the maintainers as specified here - https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md#how-to-get-prs-merged

**Comment by LuisOsta** [comment]
@hdost Thanks for the merge! Do you have any idea of when you'd release the next version of `opentelemetry-http`?

**Comment by cijothomas** [comment]
> @hdost Thanks for the merge! Do you have any idea of when you'd release the next version of `opentelemetry-http`?

we haven't planned the next release yet. But we are hoping to do monthly releases, so expect one in~2 weeks. If you want to request an earlier one- off release, please reply here.

**Comment by LuisOsta** [comment]
@cijothomas If an earlier release is possible that would be great! The sooner we can move off openssl the better

**Comment by cijothomas** [comment]
> @cijothomas If an earlier release is possible that would be great! The sooner we can move off openssl the better

ack.

**Comment by cijothomas** [comment]
https://github.com/open-telemetry/opentelemetry-rust/pull/1647 release PR

**Comment by BenoitRanque** [comment]
Apologies if I'm missing something here but, is this working as intended?
`reqwest` still brings in the `hyper-tls`>`native-tls`>`openssl` dependencies, unless `default-features` is set to `false`

This is relevant because one motivation to use `rustls` is to avoid the openssl compile time dependency

**Pull request merged**